### PR TITLE
GraphQL

### PR DESCRIPTION
**Pull request by zachstence**
Changes:
- Add GQL Client using Zeus
- Add GraphQL decorators to all DB entities
- Resolvers for OrganizationConiguration, SprocketConfiguration, User, and UserAuthenticationAccount
- Delete unneeded RMQ controllers

### CODE REVIEW

File: gql-client.ts [hunk 1]
```diff
@@ -0,0 +1,2972 @@
+/* eslint-disable */
```
Comment (0c370t) [inline gql-client.ts:1]: I'm not convinced we need to commit this

Comment (0c370t) [reply gql-client.ts:1]: Actually this isn't in the gql-client directory, we 100% don't need to commit this

File: migrations/1640558556329-Automigration.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import {MigrationInterface, QueryRunner} from "typeorm";
+
+export class Automigration1640558556329 implements MigrationInterface {
+    name = 'Automigration1640558556329'
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`ALTER TABLE "public"."organization_profile" ADD "websiteUrl" character varying NOT NULL DEFAULT ''`);
```
Comment (0c370t) [inline migrations/1640558556329-Automigration.ts:7]: Why are there migrations here?

File: src/app.module.ts [hunk 3]
```diff
@@ -1,19 +1,22 @@
...
 import {GlobalModule} from "./global.module";
 import {IdentityModule} from "./identity/identity.module";
 import {OrganizationModule} from "./organization/organization.module";
-import {UtilityModule} from "./utility/utility.module";
 
 @Module({
     imports: [GlobalModule,
+        GraphQLModule.forRoot({
+            autoSchemaFile: true,
+            playground: true,
```
Comment (0c370t) [inline src/app.module.ts:15]: We probably want this to be configurable so we don't expose it on the production service

File: src/configuration/organization-configuration/organization-confguration.resolvers.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import {OrganizationConfigurationService} from "./organization-configuration.service";
+import {OrganizationConfiguration} from "./organization-configuration.types";
+
+@Resolver(() => OrganizationConfiguration)
+export class OrganizationConfigurationResolver {
+    constructor(private readonly ocService: OrganizationConfigurationService) {}
+
+    @Query(() => [OrganizationConfiguration])
+    async getOrganizationConfigurations(
+        @Args("organizationId", {nullable: true}) organizationId?: number,
```
Comment (0c370t) [inline src/configuration/organization-configuration/organization-confguration.resolvers.ts:14]: OrganizationId should not be nullable here

File: src/configuration/organization-configuration/organization-configuration.service.ts [hunk 5]
```diff
@@ -1,24 +1,42 @@
...
-        private allowedValueRepository: Repository<OrganizationConfigurationAllowedValue>,
-        @InjectRepository(OrganizationConfigurationValue)
-        private valueRepository: Repository<OrganizationConfigurationValue>,
+        @InjectRepository(Organization) private organizationRepository: Repository<Organization>,
+        @InjectRepository(OrganizationConfigurationKey) private keyRepository: Repository<OrganizationConfigurationKey>,
+        @InjectRepository(OrganizationConfigurationAllowedValue) private allowedValueRepository: Repository<OrganizationConfigurationAllowedValue>,
+        @InjectRepository(OrganizationConfigurationValue) private valueRepository: Repository<OrganizationConfigurationValue>,
     ) {}
 
+    async getOrganizationConfigurations(organizationId?: number, keyCode?: string): Promise<OrganizationConfiguration[]> {
```
Comment (0c370t) [inline src/configuration/organization-configuration/organization-configuration.service.ts:21]: ditto for no nullable organizationId

File: src/configuration/verbiage/verbiage.service.ts [hunk 6]
```diff
@@ -56,7 +55,8 @@ export class VerbiageService {
      * @param query A query to search for matching Verbiage.
      * @returns The verbiage that matches the query.
      */
-    async getVerbiage(query: VerbiageQuery): Promise<string> {
+    // TODO type this
+    async getVerbiage(query: {organizationId: number; code: string;}): Promise<string> {
```
Comment (0c370t) [inline src/configuration/verbiage/verbiage.service.ts:59]: Type this

File: src/game/game-mode/game-mode.service.ts [hunk 7]
```diff
@@ -13,7 +13,8 @@ export class GameModeService {
         return this.gameModeRepository.findOneOrFail(id);
     }
 
-    async getGameModes(query: GameModesQuery): Promise<GameMode[]> {
+    // TODO type this
+    async getGameModes(query: FindManyOptions<GameMode>): Promise<GameMode[]> {
```
Comment (0c370t) [inline src/game/game-mode/game-mode.service.ts:17]: Type this?

File: src/game/game/game.service.ts [hunk 8]
```diff
@@ -13,7 +13,8 @@ export class GameService {
         return this.gameRepository.findOneOrFail(id);
     }
 
-    async getGames(query: GamesQuery): Promise<Game[]> {
+    // TODO type this
+    async getGames(query: FindManyOptions<Game>): Promise<Game[]> {
```
Comment (0c370t) [inline src/game/game/game.service.ts:17]: Type this?

File: src/organization/member-platform-account/member-platform-account.service.ts [hunk 9]
```diff
@@ -19,7 +19,8 @@ export class MemberPlatformAccountService {
         return this.memberPlatformAccountRepository.findOneOrFail(id);
     }
 
-    async getMemberPlatformAccounts(query: CoreInput<typeof CoreEndpoint.MemberPlatformAccount_Query>): Promise<MemberPlatformAccount[]> {
+    // TODO type this
+    async getMemberPlatformAccounts(query: FindConditions<MemberPlatformAccount>): Promise<MemberPlatformAccount[]> {
```
Comment (0c370t) [inline src/organization/member-platform-account/member-platform-account.service.ts:23]: Type this?

File: src/organization/organization/organization.service.ts [hunk 10]
```diff
@@ -33,15 +33,17 @@ export class OrganizationService {
      * @retusn The organization with the given id, if found.
      */
     async getOrganizationById(id: number): Promise<Organization> {
-        return this.organizationRepository.findOneOrFail(id);
+        // TODO how to determine when to populate things? Can we do it based on GraphQL queries? Maybe inside our wrapping functions
```
Comment (0c370t) [inline src/organization/organization/organization.service.ts:36]: Should most likely be done on GraphQL Resolvers, or manually when needed for a service call

File: src/organization/pronouns/pronouns.service.ts [hunk 11]
```diff
@@ -36,10 +35,11 @@ export class PronounsService {
         return pronouns;
     }
 
-    async getPronouns(query: PronounsQuery): Promise<Pronouns[]> {
+    // TODO type this
+    async getPronouns(query: {organizationId: number;}): Promise<Pronouns[]> {
```
Comment (0c370t) [inline src/organization/pronouns/pronouns.service.ts:39]: Type this?

**Review by 0c370t — CHANGES_REQUESTED** [review]

File: gql-client.ts (hunk 1 above)
Comment (zachstence) [reply gql-client.ts]: f7a0a4f

File: src/app.module.ts (hunk 3 above)
Comment (zachstence) [reply src/app.module.ts]: 69679d7

File: src/configuration/organization-configuration/organization-confguration.resolvers.ts (hunk 4 above)
Comment (zachstence) [reply src/configuration/organization-configuration/organization-confguration.resolvers.ts]: 0f5420e

File: src/configuration/organization-configuration/organization-configuration.service.ts (hunk 5 above)
Comment (zachstence) [reply src/configuration/organization-configuration/organization-configuration.service.ts]: 0f5420e

File: src/game/game-mode/game-mode.service.ts (hunk 7 above)
Comment (zachstence) [reply src/game/game-mode/game-mode.service.ts:16]: 0d6e667

File: src/game/game/game.service.ts (hunk 8 above)
Comment (zachstence) [reply src/game/game/game.service.ts:16]: 0d6e667

File: src/organization/member-platform-account/member-platform-account.service.ts (hunk 9 above)
Comment (zachstence) [reply src/organization/member-platform-account/member-platform-account.service.ts:22]: 0d6e667

File: src/organization/pronouns/pronouns.service.ts (hunk 11 above)
Comment (zachstence) [reply src/organization/pronouns/pronouns.service.ts:39]: 0d6e667

File: src/organization/organization/organization.service.ts (hunk 10 above)
Comment (zachstence) [reply src/organization/organization/organization.service.ts]: I am brick c9c7fc7

**Pull request merged**